presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -246,12 +249,18 @@ public void addInput(Page page)
         checkState(state == State.RUNNING, "Operator is %s", state);
 
         TableCommitContext tableCommitContext = getTableCommitContext(page, tableCommitContextCodec);
-        lifespanAndStageStateTracker.update(page, tableCommitContext);
-        lifespanAndStageStateTracker.getStatisticsPagesToProcess(page, tableCommitContext).forEach(statisticsPage -> {
-            OperationTimer timer = new OperationTimer(statisticsCpuTimerEnabled);
-            statisticsAggregationOperator.addInput(statisticsPage);
-            timer.end(statisticsTiming);
-        });
+        boolean flag = lifespanAndStageStateTracker.update(page, tableCommitContext);
+        if (flag) {
+            lifespanAndStageStateTracker.getStatisticsPagesToProcess(page, tableCommitContext).forEach(statisticsPage -> {
+                OperationTimer timer = new OperationTimer(statisticsCpuTimerEnabled);
+                statisticsAggregationOperator.addInput(statisticsPage);
+                timer.end(statisticsTiming);
+            });
+        }
+        else {
+            LOG.info("The lifeSpanAndStage %s has been committed before", LifespanAndStage.fromTableCommitContext(tableCommitContext));
```

Comment:
Since this isn't actionable and no one needs to read this, this doesn't seem worth an info log. Debug level at most, but better yet nothing at all.

Reply:
Sure, the log is removed! Thanks for the suggestion.